include/swift/Option/Options.td

```diff
@@ -278,6 +278,13 @@ def emit_tbd_path_EQ : Joined<["-"], "emit-tbd-path=">,
 def serialize_diagnostics : Flag<["-"], "serialize-diagnostics">,
   Flags<[FrontendOption, NoInteractiveOption, DoesNotAffectIncrementalBuild]>,
   HelpText<"Serialize diagnostics in a binary format">;
+def serialize_diagnostics_path : Separate<["-"], "serialize-diagnostics-path">,
+  Flags<[FrontendOption, DoesNotAffectIncrementalBuild, ArgumentIsPath]>,
```

Comment:
Can you mark these "NoBatchOption" so that they're only allowed by the interpreter, and then can you add a test that shows them being rejected by swiftc?

Reply:
Fixed.